File: examples/java/src/ChsNluExample1.java
import tencent.ai.texsmart.*;

public class ChsNluExample1 {
	public static void main(String[] args) {
		String userDir = System.getProperty("user.dir");
		String dataDir = args.length >= 1 ? args[0] : userDir + "/../../data/nlu/kb/";

		System.out.println("Creating and initializing the NLU engine...");
		NluEngine engine = new NluEngine();
		int workerCount = 4;
		boolean ret = engine.init(dataDir, workerCount);
		if(!ret) {
			System.out.println("Failed to initialize the engine");
			return;
		}

		System.out.println("=== 解析一个中文句子 ===");
		String text = "上个月30号，南昌王先生在自己家里边看流浪地球边吃煲仔饭";
		NluOutput output = engine.parseText(text);

		System.out.println("细粒度分词:");
		for(NluOutput.Term word : output.words()) {
			System.out.printf("\t%s\t(%d,%d)\t%s\t%d\n", word.str, word.offset, word.len, word.tag, word.freq);
		}

		System.out.println("粗粒度分词:");
		for(NluOutput.Term phrase : output.phrases()) {
			System.out.printf("\t%s\t(%d,%d)\t%s\t%d\n", phrase.str, phrase.offset, phrase.len, phrase.tag, phrase.freq);
		}

		System.out.println("命名实体识别（NER）:");
		for(NluOutput.Entity ent : output.entities()) {
			String typeStr = String.format("(%s,%s,%d,%s)", ent.type.name, ent.type.i18n, ent.type.flag, ent.type.path);
			System.out.printf("\t%s\t(%d,%d)\t%s\t%s\n", ent.str, ent.offset, ent.len, typeStr, ent.meaning);
		}
		
		engine.clear();
	}
}

File: examples/python/chs_nlu_example1.py
#!/usr/bin/env python
#encoding=utf-8
import sys
import os.path
module_dir = os.path.dirname(os.path.realpath(__file__))
sys.path.append(module_dir+'/../../lib/')
from tencent_ai_texsmart import *

print('##################################################')
print('# Example-1: Parsing text (with default options)')
print('##################################################')

print('Creating and initializing the NLU engine...')
engine = NluEngine(module_dir + '/../../data/nlu/kb/', 1)

print(u'=== 解析一个中文句子 ===')
output = engine.parse_text(u"上个月30号，南昌王先生在自己家里边看流浪地球边吃煲仔饭")
print(u'细粒度分词:')
for item in output.words():
    print(u'\t{0}\t{1}\t{2}\t{3}\t{4}'.format(item.str, item.offset, item.len, item.tag, item.freq))
print(u'粗粒度分词:')
for item in output.phrases():
    print(u'\t{0}\t{1}\t{2}\t{3}\t{4}'.format(item.str, item.offset, item.len, item.tag, item.freq))
print(u'命名实体识别（NER）:')
for entity in output.entities():
    type_str = u'({0},{1},{2},{3})'.format(entity.type.name, entity.type.i18n, entity.type.flag, entity.type.path)
    print(u'\t{0}\t({1},{2})\t{3}\t{4}'.format(entity.str, entity.offset, entity.len, type_str, entity.meaning))
